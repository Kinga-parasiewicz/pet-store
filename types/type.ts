import { Dispatch, FormEvent } from "react";

export interface Pet {
   id:number;
   name:string;
   photoUrls?:string;
   category: string;
   status: Status;
}


export type Status = " "| "available" |"pending" | "sold"


export type PetResponse = {
  state: string,
  pets: Pet[] | null;
}

export type AppProps = {
  title?: string;
  buttonName?: string;
  confirmButton?: string;
  handleSubmit?:(e: FormEvent<HTMLFormElement>) => void
  setBody?:Dispatch<React.SetStateAction<Pet|undefined>>;
  body?:Pet;
}