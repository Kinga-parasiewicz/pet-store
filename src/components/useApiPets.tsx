import { useState, useEffect } from "react";
import axios from "axios";
import SERVER_BASE_URL from "./../config";
import { PetResponse } from "types/type";

export const useApiPets = () => {
  const [petsData, setPetsData] = useState<PetResponse>({
    state: "loading",
    pets: [],
  });

  const getExamplePets = async () => {
    await axios
      .get<PetResponse>(`${SERVER_BASE_URL}/examplePets.json`)
      .then((response) => setPetsData(response.data))
      .catch(() => setPetsData({ state: "error", pets: [] }));
  };

  useEffect(() => {
    getExamplePets();
    setTimeout(getExamplePets, 2000);
  }, []);

  return petsData;
};
