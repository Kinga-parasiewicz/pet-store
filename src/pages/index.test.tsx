import renderer from 'react-test-renderer'
import Home from '.'
import { PetResponse } from 'types/type'
import { useApiPets } from './../components/useApiPets'


jest.mock("./../components/useApiPets")
const mockUseClientRect = useApiPets as jest.MockedFunction<typeof useApiPets>
const mockData: PetResponse = {
    state: 'loaded',
    pets: [
        {
            id: 3,
            category: 'Cat',
            name: 'Leo',
            photoUrls: '/cat-01.jpeg',
            status: 'available',
        },
    ],
}

describe('Table is loaded', () => {
    it('renders table', () => {
        mockUseClientRect.mockReturnValue(mockData)
        const domTree = renderer.create(<Home />).toJSON()
        expect(domTree).toMatchSnapshot()
    })
})
