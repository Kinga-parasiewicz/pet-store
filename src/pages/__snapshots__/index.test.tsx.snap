// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table is loaded renders table 1`] = `
<div
  className="mx-auto md:px-8 max-w-[1440px]"
>
  <header
    className="bg-blue-700 h-20 flex"
  >
    <h1
      className=" px-4 text-white font-bold mx-4"
      role="heading"
    >
      PetStore
    </h1>
    <div
      className="flex bg-blue-700 h-20 items-center gap-4"
    >
      <a
        className="text-white text-xl  font-bold "
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Home
      </a>
      <a
        className="text-white text-xl font-bold"
        href="/pet"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        Pet
      </a>
    </div>
  </header>
  <div>
    <h2
      className="font-bold text-blue-700"
    >
      Pets
    </h2>
    <table
      className="table-fixed text-center"
    >
      <thead>
        <tr
          className="text-white bg-blue-700 h-8"
        >
          <th>
            Photo
          </th>
          <th>
            Pet Name
          </th>
          <th>
            Category
          </th>
          <th>
            Status
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            width={200}
          >
            <img
              alt="Leo"
              className="pr-4"
              src="/cat-01.jpeg"
            />
          </td>
          <td>
            Leo
          </td>
          <td>
            Cat
          </td>
          <td>
            available
          </td>
          <td
            width={300}
          >
            <button
              onClick={[Function]}
            >
              delete 
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
